auditlog/cid.py

```diff
@@ -0,0 +1,60 @@
+from contextvars import ContextVar
+from typing import Optional
+
+from django.conf import settings
+from django.http import HttpRequest
+from django.utils.module_loading import import_string
+
+correlation_id = ContextVar("auditlog_correlation_id", default=None)
+
+
+def set_cid(request: Optional[HttpRequest] = None) -> None:
+    """
+    A function to read the cid from a request.
+    If the header is not in the request, then we set it to `None`.
+
+    Note: we look for the header in `request.headers` and `request.META`.
+
+    :param request: The request to get the cid from.
+    :return: None
+    """
+    cid = None
+    header = settings.AUDITLOG_CID_HEADER
+
+    if header and request and header in request.headers or header in request.META:
+        cid = request.headers.get(header)
+
+    # in theory, we shouldn't have to set the cid back to None because each request should be in a new thread.
+    # however, this was causing some tests to fail.
```

Comment:
Didn't get this comment here. do we really set cid here to None? what if the cid present in header

Reply:
The idea is that I had to move setting the contextvar outside of the if statement. Which means that we'll be explicitly setting cid to None if the cid is not in the header.

Django requests are multi threaded but the tests are not.

Reply:
Yeah, I see.  but I didn't get the explanation from your comment.
You may need to phrase it better.

Reply:
How about:
"Ideally, this line should be nested inside the if statement. However, because the tests do not run requests in multiple threads, we have to always set the value of the cid, even if the request does not have the header present, in which case it will be set to None"

Reply:
Great!

Reply:
```suggestion
    # Ideally, this line should be nested inside the if statement. However, because
    # the tests do not run requests in multiple threads, we have to always set the
    # value of the cid, even if the request does not have the header present, in which
    # case it will be set to None
```

Reply:
pushed